src/Microsoft.Performance.SDK.Runtime/Options/PluginOptionsRegistry.cs

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Performance.SDK.Options;
+using Microsoft.Performance.SDK.Runtime.Options.Serialization.DTO;
+
+namespace Microsoft.Performance.SDK.Runtime.Options;
+
+/// <summary>
+///     Represents a registry of <see cref="PluginOption"/> instances.
+/// </summary>
+public sealed class PluginOptionsRegistry
+{
+    private readonly object mutex = new();
+    private readonly Dictionary<Guid, PluginOption> optionByGuid = new();
+
+    /// <summary>
+    ///     Represents a class that can provide <see cref="PluginOption"/> instances to register to this class. Do not
+    ///     implement this interface directly. Instead, use public methods on <see cref="PluginOptionsSystem"/> to register
+    ///     options from supported sources.
+    /// </summary>
+    internal interface Provider
```

Comment:
>Provider [](http://example.com/codeflow?start=23&length=8)

an interface that doesn't start with 'I'. Is that something we really want?

Reply:
I'll change it, but I left it off since this is nested in a class. `TestPluginOptionsRegistryProvider : PluginOptionsRegistry.Provider` looked better to me than `TestPluginOptionsRegistryProvider : PluginOptionsRegistry.IProvider` 🤷

Reply:
Thanks, I think the consistency will make it easier to read.